src/autoformat.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import AutoformatEngine from './autoformatengine.js';
+import Feature from '../core/feature.js';
+import HeadingEngine from '../heading/headingengine.js';
+import ListEngine from '../list/listengine.js';
+
+/**
+ * The autoformat feature. Looks for predefined regular expressions and converts inserted text accordingly.
+ *
+ * @memberOf autoformat
+ * @extends core.Feature
+ */
+export default class Autoformat extends Feature {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ HeadingEngine, ListEngine ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+
+		if ( editor.commands.has( 'bulletedList' ) ) {
+			new AutoformatEngine( editor, /^[\*\-]\s$/, 'bulletedList' );
```

Comment:
I remember we talked f2f about this, but to be sure: do we really need `\s` instead of just a space character?